File: uikit/src/main/java/com/cometchat/pro/uikit/Sticker/StickerView.java
package com.cometchat.pro.uikit.Sticker;

import android.content.Context;
import android.content.res.ColorStateList;
import android.content.res.TypedArray;
import android.graphics.Color;
import android.graphics.drawable.Drawable;
import android.os.Bundle;
import android.os.Parcelable;
import android.util.AttributeSet;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.RelativeLayout;

import androidx.core.graphics.drawable.DrawableCompat;
import androidx.fragment.app.FragmentActivity;
import androidx.viewpager.widget.ViewPager;

import com.bumptech.glide.Glide;
import com.cometchat.pro.uikit.R;
import com.cometchat.pro.uikit.Settings.UISettings;
import com.cometchat.pro.uikit.Sticker.listener.StickerClickListener;
import com.cometchat.pro.uikit.Sticker.model.Sticker;
import com.google.android.material.tabs.TabLayout;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;

import adapter.StickerTabAdapter;
import utils.Utils;


public class StickerView extends RelativeLayout implements StickerClickListener {

    private Context context;

    private ViewPager viewPager;

    private TabLayout tabLayout;

    private String Id;

    private String type;

    private AttributeSet attrs;

    private StickerTabAdapter adapter;

    private HashMap<String,List<Sticker>> stickerMap = new HashMap<>();

    private StickerClickListener stickerClickListener;

    public StickerView(Context context) {
        super(context);
        this.context = context;
        initViewComponent(context,null,-1,-1);
    }

    public StickerView(Context context, AttributeSet attrs) {
        super(context, attrs);
        this.attrs = attrs;
        this.context = context;
        initViewComponent(context,attrs,-1,-1);
    }

    public StickerView(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
        this.context = context;
        this.attrs = attrs;
        initViewComponent(context,attrs,defStyleAttr,-1);
    }


    private void initViewComponent(Context context,AttributeSet attributeSet,int defStyleAttr,int defStyleRes){

        View view =View.inflate(context, R.layout.cometchat_sticker_view,null);

        TypedArray a = getContext().getTheme().obtainStyledAttributes(attributeSet, R.styleable.SharedMediaView, 0, 0);
        addView(view);


        if (type!=null) {
            viewPager = this.findViewById(R.id.viewPager);
            tabLayout = view.findViewById(R.id.tabLayout);
            adapter = new StickerTabAdapter(context,((FragmentActivity)context).getSupportFragmentManager());
            for (String str : stickerMap.keySet()) {
                Bundle bundle = new Bundle();
                bundle.putString("Id",Id);
                bundle.putString("type",type);
                StickerFragment stickersFragment = new StickerFragment();
                bundle.putParcelableArrayList("stickerList", (ArrayList<? extends Parcelable>) stickerMap.get(str));
                stickersFragment.setArguments(bundle);
                stickersFragment.setStickerClickListener(stickerClickListener);
                adapter.addFragment(stickersFragment, str,stickerMap.get(str).get(0).getUrl());
            }
            viewPager.setAdapter(adapter);
            tabLayout.setupWithViewPager(viewPager);

            for (int i=0;i<tabLayout.getTabCount();i++) {
                tabLayout.getTabAt(i).setCustomView(createTabItemView(adapter.getPageIcon(i)));
            }
            if (UISettings.getColor()!=null) {
                Drawable wrappedDrawable = DrawableCompat.wrap(getResources().
                        getDrawable(R.drawable.tab_layout_background_active));
                DrawableCompat.setTint(wrappedDrawable, Color.parseColor(UISettings.getColor()));
                tabLayout.getTabAt(tabLayout.getSelectedTabPosition()).view.setBackground(wrappedDrawable);
                tabLayout.setSelectedTabIndicatorColor(Color.parseColor(UISettings.getColor()));
            } else {
                tabLayout.getTabAt(tabLayout.getSelectedTabPosition()).
                        view.setBackgroundColor(getResources().getColor(R.color.colorPrimary));
                tabLayout.setSelectedTabIndicatorColor(getResources().getColor(R.color.colorPrimary));
            }
            tabLayout.addOnTabSelectedListener(new TabLayout.OnTabSelectedListener() {
                @Override
                public void onTabSelected(TabLayout.Tab tab) {
                    if (UISettings.getColor()!=null) {
                        Drawable wrappedDrawable = DrawableCompat.wrap(getResources().
                                getDrawable(R.drawable.tab_layout_background_active));
                        DrawableCompat.setTint(wrappedDrawable, Color.parseColor(UISettings.getColor()));
                        tab.view.setBackground(wrappedDrawable);
                    }
                    else
                        tab.view.setBackgroundColor(getResources().getColor(R.color.colorPrimary));
                }

                @Override
                public void onTabUnselected(TabLayout.Tab tab) {
                    tab.view.setBackgroundColor(getResources().getColor(android.R.color.transparent));
                }

                @Override
                public void onTabReselected(TabLayout.Tab tab) {

                }
            });


        }
    }

    private View createTabItemView(String imgUri) {
        ImageView imageView = new ImageView(context);
        TabLayout.LayoutParams params = new TabLayout.LayoutParams(72, 72);
        imageView.setLayoutParams(params);
        Glide.with(context).load(imgUri).into(imageView);
        return imageView;
    }

    public void setStickerClickListener(StickerClickListener stickerClickListener) {
        this.stickerClickListener = stickerClickListener;
    }
    public void setData(String uid,String receiverType,HashMap<String,List<Sticker>> stickers) {
        this.Id = uid;
        this.type = receiverType;
        this.stickerMap = stickers;
        reload();
    }


    public void reload() {
        initViewComponent(context,null,-1,-1);
    }

    @Override
    public void onClickListener(Sticker sticker) {
        stickerClickListener.onClickListener(sticker);
    }
}

File: uikit-kotlin/src/main/java/com/cometchat/pro/uikit/UI_Components/shared/cometchatStickers/StickerView.kt
package com.cometchat.pro.uikit.ui_components.shared.cometchatStickers

import com.cometchat.pro.uikit.ui_components.shared.cometchatStickers.adapter.StickerTabAdapter
import android.content.Context
import android.os.Bundle
import android.os.Parcelable
import android.util.AttributeSet
import android.view.View
import android.widget.FrameLayout
import android.widget.ImageView
import android.widget.RelativeLayout
import androidx.fragment.app.FragmentActivity
import androidx.viewpager.widget.ViewPager
import com.bumptech.glide.Glide
import com.cometchat.pro.uikit.R
import com.cometchat.pro.uikit.ui_components.shared.cometchatStickers.listener.StickerClickListener
import com.cometchat.pro.uikit.ui_components.shared.cometchatStickers.model.Sticker
import com.google.android.material.tabs.TabLayout
import java.util.*

class StickerView : RelativeLayout, StickerClickListener {

    private var c: Context? = null

    private var viewPager: ViewPager? = null

    private var tabLayout: TabLayout? = null

    private var Id: String? = null

    private var type: String? = null

    private var attrs: AttributeSet? = null

    private var adapter: StickerTabAdapter? = null

    private var stickerMap = java.util.HashMap<String?, MutableList<Sticker>?>()

    private var stickerClickListener: StickerClickListener? = null

    constructor(context: Context?) : super(context){
        c = context
        initViewComponent(context, null, -1, -1)
    }
    constructor(context: Context?, attrs: AttributeSet?) : super(context, attrs){
        this.attrs = attrs
        c = context
        initViewComponent(context, attrs, -1, -1)
    }
    constructor(context: Context?, attrs: AttributeSet?, defStyleAttr: Int) : super(context, attrs, defStyleAttr){
        c = context
        this.attrs = attrs
        initViewComponent(context, attrs, defStyleAttr, -1)
    }

    private fun initViewComponent(context: Context?, attributeSet: AttributeSet?, defStyleAttr: Int, i: Int) {
        var view = inflate(context, R.layout.cometchat_sticker_view, null)
        val a = getContext().theme.obtainStyledAttributes(attributeSet, R.styleable.SharedMediaView, 0, 0)
        addView(view)

        if (type != null) {
            viewPager = findViewById(R.id.viewPager)
            tabLayout = view.findViewById(R.id.tabLayout)
            adapter = StickerTabAdapter(context!!, (context as FragmentActivity).supportFragmentManager)
        for (str in stickerMap.keys) {
            val bundle = Bundle()
            bundle.putString("Id", Id)
            bundle.putString("type", type)
            val stickersFragment = StickerFragment()
            bundle.putParcelableArrayList("stickerList", stickerMap[str] as ArrayList<out Parcelable?>?)
            stickersFragment.setArguments(bundle)
            stickersFragment.setStickerClickListener(stickerClickListener)
            adapter?.addFragment(stickersFragment, str, stickerMap[str]!![0].url)
        }
        viewPager?.adapter = adapter
        tabLayout?.setupWithViewPager(viewPager)

        for (i in 0 until tabLayout!!.tabCount) {
            tabLayout!!.getTabAt(i)?.setCustomView(createTabItemView(adapter?.getPageIcon(i)!!))
        }

        tabLayout!!.getTabAt(tabLayout!!.selectedTabPosition)!!.view.setBackgroundColor(resources.getColor(R.color.colorPrimary))
        tabLayout!!.setSelectedTabIndicatorColor(resources.getColor(R.color.colorPrimary))

        tabLayout!!.addOnTabSelectedListener(object : TabLayout.OnTabSelectedListener {
            override fun onTabSelected(tab: TabLayout.Tab) {
                tab.view.setBackgroundColor(resources.getColor(R.color.colorPrimary))
            }

            override fun onTabUnselected(tab: TabLayout.Tab) {
                tab.view.setBackgroundColor(resources.getColor(android.R.color.transparent))
            }

            override fun onTabReselected(tab: TabLayout.Tab) {}
        })
    }
    }

    private fun createTabItemView(imgUri: String): View? {
        val imageView = ImageView(context)
        val params = FrameLayout.LayoutParams(72, 72)
        imageView.layoutParams = params
        Glide.with(context).load(imgUri).into(imageView)
        return imageView
    }

    fun setStickerClickListener(stickerClickListener: StickerClickListener?) {
        this.stickerClickListener = stickerClickListener
    }

    public fun setData(uid: String, receiverType: String, stickers: java.util.HashMap<String?, MutableList<Sticker>?>){
        Id = uid
        type = receiverType
        stickerMap = stickers
        reload()
    }

    fun reload() {
        initViewComponent(context, null, -1, -1)
    }

    override fun onClickListener(sticker: Sticker?) {
        stickerClickListener!!.onClickListener(sticker)
    }
}